core/src/transfer/moonlight.rs

```diff
@@ -173,6 +173,26 @@ impl Transaction {
         }
     }
 
+    /// Returns whether the transaction is a self-transfer.
+    ///
+    /// A self-transfer is defined as a moonlight transaction where the sender
+    /// and receiver are the same and a normal transfer is being made.
+    #[must_use]
+    pub fn is_self_transfer(&self) -> bool {
```

Comment:
Shouldn't we check if the change receiver is the same too?

Otherwise we are spoofing prone